libs/entitlement/bridge/src/main/java/org/elasticsearch/entitlement/bridge/EntitlementChecker.java

```diff
@@ -371,6 +382,147 @@ public interface EntitlementChecker {
 
     void check$sun_net_www_URLConnection$getContentLength(Class<?> callerClass, java.net.URLConnection that);
 
+    void check$sun_net_www_protocol_ftp_FtpURLConnection$connect(Class<?> callerClass, java.net.URLConnection that);
+
+    void check$sun_net_www_protocol_ftp_FtpURLConnection$getInputStream(Class<?> callerClass, java.net.URLConnection that);
+
+    void check$sun_net_www_protocol_ftp_FtpURLConnection$getOutputStream(Class<?> callerClass, java.net.URLConnection that);
+
+    void check$sun_net_www_protocol_http_HttpURLConnection$$openConnectionCheckRedirects(Class<?> callerClass, java.net.URLConnection c);
+
+    void check$sun_net_www_protocol_http_HttpURLConnection$connect(Class<?> callerClass, java.net.HttpURLConnection that);
+
```

Comment:
what about `doTunneling`?

Reply:
and looks like this is yet another case we haven't handled yet :/
```
protected HttpClient getNewHttpClient(URL url, Proxy p, int connectTimeout)
        throws IOException {
        return HttpClient.New(url, p, connectTimeout, this);
    }
```

Reply:
I added those and then removed them because they are "internal" - they are public but in non-exported classes.
(I realized because I was not able to test them :D )

Reply:
I was just about to say, this is probably not exported 👍